xdsl/dialects/llvm.py

```diff
@@ -95,6 +97,66 @@ def is_typed(self):
         return not isinstance(self.type, NoneAttr)
 
 
+@irdl_op_definition
+class GEPOp(Operation):
+    name = "llvm.getelementptr"
+
+    ptr: Annotated[Operand, LLVMPointerType]
+
+    ssa_indices: Annotated[VarOperand, IntegerType]
+
+    elem_type: OptOpAttr[Attribute]
+
+    rawConstantIndices: OpAttr[DenseArrayBase]
+
+    inbounds: OptOpAttr[UnitAttr]
+
+    result: Annotated[OpResult, LLVMPointerType]
+
+    @staticmethod
+    def get(ptr: SSAValue | Operation,
+            result_type: Attribute
+            | LLVMPointerType = LLVMPointerType.opaque(),
+            indices: list[int] | None = None, #Here we are assuming the indices follow the MLIR standard (min int where the SSA value should be used)
+            ssa_indices: list[SSAValue | Operation] | None = None,
+            inbounds: bool = False,
+            pointee_type: Attribute | None = None):
+
+        if indices is None:
+            raise ValueError('llvm.getelementptr must have a pointer.')
```

Comment:
This looks out of date to me. Shouldn't this be something like "needs indices" or something?

Reply:
Yes...., I was half asleep when I added these lol